RecoTracker/ConversionSeedGenerators/python/PhotonConversionTrajectorySeedProducerFromSingleLeg_cfi.py

```diff
@@ -20,14 +25,7 @@
                                                   SeedComparitorPSet = cms.PSet(
                                                       ComponentName = cms.string('none')
                                                       ),
-                                                  ClusterCheckPSet = cms.PSet(
-                                                      PixelClusterCollectionLabel = cms.InputTag("siPixelClusters"),
-                                                      MaxNumberOfCosmicClusters = cms.uint32(400000),
-                                                      doClusterCheck = cms.bool(True),
-                                                      ClusterCollectionLabel = cms.InputTag("siStripClusters"),
-                                                      MaxNumberOfPixelClusters = cms.uint32(40000),
-                                                      cut = cms.string("strip < 400000 && pixel < 40000 && (strip < 50000 + 10*pixel) && (pixel < 5000 + 0.1*strip)"),
-                                                      ),
+                                                  ClusterCheckPSet = CommonClusterCheckPSet,
```

Comment:
In SeedGeneratorFromRegionHitsEDProducer_cfi.py
https://cmssdt.cern.ch/lxr/source/RecoTracker/TkSeedGenerator/python/SeedGeneratorFromRegionHitsEDProducer_cfi.py#0030
doClusterCheck is stll true.

Therefore in this PR you are not changing it. Did I miss something, or you should set it to false? Up to now the photonConvTrajSeedFromSingleLeg configuration in this PR is identical to what it is now in the release

Reply:
@perrotta `doClusterCheck` is set to false for 2017 with `phase1Pixel` sub-era in line 42
https://cmssdt.cern.ch/lxr/source/RecoTracker/TkSeedGenerator/python/SeedGeneratorFromRegionHitsEDProducer_cfi.py#0042

Reply:
Thank you @makortel 

However, if I make a edmConfigDump of the step3 config I extract what follows, which confirms my finding that the results of wf 10224 from this PR when the addition of the new layer pairs in the seed is removed are the same as the ones in the baseline

process.photonConvTrajSeedFromSingleLeg = cms.EDProducer("PhotonConversionTrajectorySeedProducerFromSingleLeg",
    ClusterCheckPSet = cms.PSet(
        ClusterCollectionLabel = cms.InputTag("siStripClusters"),
        MaxNumberOfCosmicClusters = cms.uint32(400000),
        MaxNumberOfPixelClusters = cms.uint32(40000),
        PixelClusterCollectionLabel = cms.InputTag("siPixelClusters"),
        cut = cms.string('strip < 400000 && pixel < 40000 && (strip < 50000 + 10*pixel) && (pixel < 5000 + 0.1*strip)'),
        doClusterCheck = cms.bool(True)
    ),
    DoxcheckSeedCandidates = cms.bool(False),
    OrderedHitsFactoryPSet = cms.PSet(
        SeedingLayers = cms.InputTag("convLayerPairs"),
        maxElement = cms.uint32(40000),
        maxHitPairsPerTrackAndGenerator = cms.uint32(10)
    ),
    RegionFactoryPSet = cms.PSet(
        ComponentName = cms.string('GlobalRegionProducerFromBeamSpot'),
        RegionPSet = cms.PSet(
            beamSpot = cms.InputTag("offlineBeamSpot"),
            originHalfLength = cms.double(12.0),
            originRadius = cms.double(3.0),
            precise = cms.bool(True),
            ptMin = cms.double(0.2)
        )
    ),
    SeedComparitorPSet = cms.PSet(
        ComponentName = cms.string('none')
    ),
    SeedCreatorPSet = cms.PSet(
        ComponentName = cms.string('SeedForPhotonConversion1Leg'),
        SeedMomentumForBOFF = cms.double(5.0),
        TTRHBuilder = cms.string('WithTrackAngle'),
        propagator = cms.string('PropagatorWithMaterial')
    ),
    TrackRefitter = cms.InputTag("generalTracks"),
    applyTkVtxConstraint = cms.bool(True),
    beamSpotInputTag = cms.InputTag("offlineBeamSpot"),
    maxDZSigmas = cms.double(10.0),
    maxNumSelVtx = cms.uint32(2),
    newSeedCandidates = cms.string('convSeedCandidates'),
    primaryVerticesTag = cms.InputTag("firstStepPrimaryVertices"),
    vtxMinDoF = cms.double(4),
    xcheckSeedCandidates = cms.string('xcheckSeedCandidates')
)

Reply:
By the way, both

process.photonConvTrajSeedFromQuadruplets = cms.EDProducer("PhotonConversionTrajectorySeedProducerFromQuadruplets",

process.photonConvTrajSeedFromSingleLeg = cms.EDProducer("PhotonConversionTrajectorySeedProducerFromSingleLeg",

have doClusterCheck = cms.bool(True) in the dump

Reply:
@perrotta

I just did
```
cmsrel CMSSW_9_3_0_pre4
cd CMSSW_9_3_0_pre4/src
cmsenv
git cms-merge-topic 20153
scram b
runTheMatrix -l 10024.0 -j 0
cd 10024.0*
edmConfigDump step3_RAW2DIGI_L1Reco_RECO_EI_PAT_VALIDATION_DQM.py
```
and 
```
process.photonConvTrajSeedFromSingleLeg = cms.EDProducer("PhotonConversionTrajectorySeedProducerFromSingleLeg",
    ClusterCheckPSet = cms.PSet(
        ClusterCollectionLabel = cms.InputTag("siStripClusters"),
        MaxNumberOfCosmicClusters = cms.uint32(400000),
        MaxNumberOfPixelClusters = cms.uint32(40000),
        PixelClusterCollectionLabel = cms.InputTag("siPixelClusters"),
        cut = cms.string('strip < 400000 && pixel < 40000 && (strip < 50000 + 10*pixel) && (pixel < 5000 + 0.1*strip)'),
        doClusterCheck = cms.bool(False)
    ),
    ...
```

Reply:
Perhaps I already moved to another $CMSSW_BASE when I  made the dump... I will try  (not now, as I am in train)

Reply:
Perhaps I already moved to another $CMSSW_BASE when I  made the dump... I will try  (not now, as I am in train)

Reply:
Ok, @makortel , you are right: I evidently moved away from this $CMSSW_BASE when I made the dump, and therefore the config currently in the release was picked instead. Setting the environment correctly I do see that the doClusterCheck is false, as it should